### ISSUE TITLE
Bug with visit:process command

### ISSUE THREAD

**Issue by tivyhosting** [shlinkio/shlink#237]
This is on version 1.13.1, trying to process a queue of about 500 clicks. Also, the site has cloudflare enabled, so that may be causing some issues?

```
Processing IP PHP Fatal error:  Uncaught TypeError: Argument 1 passed to Shlinkio\Shlink\Common\Service\IpApiLocationResolver::resolveIpLocation() must be of the type string, null given, called in /srv/users/lmg-gg/apps/shlink-1-13-1/module/CLI/src/Command/Visit/ProcessVisitsCommand.php on line 73 and defined in /srv/users/lmg-gg/apps/shlink-1-13-1/module/Common/src/Service/IpApiLocationResolver.php:32
Stack trace:
#0 /srv/users/lmg-gg/apps/shlink-1-13-1/module/CLI/src/Command/Visit/ProcessVisitsCommand.php(73): Shlinkio\Shlink\Common\Service\IpApiLocationResolver->resolveIpLocation(NULL)
#1 /srv/users/lmg-gg/apps/shlink-1-13-1/vendor/symfony/console/Command/Command.php(251): Shlinkio\Shlink\CLI\Command\Visit\ProcessVisitsCommand->execute(Object(Symfony\Component\Console\Input\ArgvInput), Object(Symfony\Component\Console\Output\ConsoleOutput))
#2 /srv/users/lmg-gg/apps/shlink-1-13-1/vendor/symfony/console/Application.php(886): Symfony\Component\Console\Command\Command->run(Object(Symfony\Component\Console\Input\ArgvInput), Object(Symfony\Co in /srv/users/lmg-gg/apps/shlink-1-13-1/module/Common/src/Service/IpApiLocationResolver.php on line 32
 
Fatal error: Uncaught TypeError: Argument 1 passed to Shlinkio\Shlink\Common\Service\IpApiLocationResolver::resolveIpLocation() must be of the type string, null given, called in /srv/users/lmg-gg/apps/shlink-1-13-1/module/CLI/src/Command/Visit/ProcessVisitsCommand.php on line 73 and defined in /srv/users/lmg-gg/apps/shlink-1-13-1/module/Common/src/Service/IpApiLocationResolver.php:32
Stack trace:
#0 /srv/users/lmg-gg/apps/shlink-1-13-1/module/CLI/src/Command/Visit/ProcessVisitsCommand.php(73): Shlinkio\Shlink\Common\Service\IpApiLocationResolver->resolveIpLocation(NULL)
#1 /srv/users/lmg-gg/apps/shlink-1-13-1/vendor/symfony/console/Command/Command.php(251): Shlinkio\Shlink\CLI\Command\Visit\ProcessVisitsCommand->execute(Object(Symfony\Component\Console\Input\ArgvInput), Object(Symfony\Component\Console\Output\ConsoleOutput))
#2 /srv/users/lmg-gg/apps/shlink-1-13-1/vendor/symfony/console/Application.php(886): Symfony\Component\Console\Command\Command->run(Object(Symfony\Component\Console\Input\ArgvInput), Object(Symfony\Co in /srv/users/lmg-gg/apps/shlink-1-13-1/module/Common/src/Service/IpApiLocationResolver.php on line 32
```

**Comment by acelaya** [comment]
It seems that the IP is not properly parsed from the request.
According to the docs (https://support.cloudflare.com/hc/en-us/articles/200170986-How-does-Cloudflare-handle-HTTP-Request-headers-), cloudflare is working as a reverse proxy and sending the X-Forwarded-For header, so maybe there's an issue on the parsing logic.

I'm going to use this https://github.com/akrabat/ip-address-middleware to avoid any missing IP, instead of trying to come with a proper logic myself.

**Comment by tivyhosting** [comment]
@acelaya Is this a rough ETA on this fix?

**Comment by acelaya** [comment]
I will try to have something done by tomorrow in the afternoon

**Comment by acelaya** [comment]
@tivyhosting could you confirm if there's any `null` entries in the `remote_addr` column of your `visits` table? (If you don't know how to check that, let me know)

**Comment by tivyhosting** [comment]
The NULL remote_addr entries span across multiple short_urls

**Comment by tivyhosting** [comment]
@acelaya The NULL remote_addr entries span across multiple short_urls

**Comment by acelaya** [comment]
It's hard to say what could be happening, but I would say it has something to do with the fact that, in this case, Shlink is behind Cloudflare.

The fix is mostly implemented here https://github.com/shlinkio/shlink/pull/238, including two things mainly:

* Make sure `null` addresses are just skipped to prevent the Fatal error
* Use https://github.com/akrabat/ip-address-middleware in order to determine the remote address, which does more checks than previous shlink custom implementation.

I will merge it (as soon as Scrutinizer wants to finish the build...) and you can try to build a dist file from master ([Installation](https://github.com/shlinkio/shlink#installation) -> **Building from sources**)

**Comment by acelaya** [comment]
I have just released Shlink v1.13.2 https://github.com/shlinkio/shlink/releases/tag/v1.13.2
